packages/server/src/db/linkedRows/index.ts

```diff
@@ -208,7 +215,32 @@ export async function squashLinksToPrimaryDisplay(
         const linkedTable = await getLinkedTable(linkTblId, linkedTables)
         const obj: any = { _id: link._id }
         if (linkedTable?.primaryDisplay && link[linkedTable.primaryDisplay]) {
-          obj.primaryDisplay = link[linkedTable.primaryDisplay]
+          const fieldSchema = linkedTable.schema[linkedTable.primaryDisplay]
+          switch (fieldSchema.type) {
+            case FieldType.BB_REFERENCE:
+              const subtype = fieldSchema.subtype as FieldSubtype
+              switch (subtype) {
+                case FieldSubtype.USER:
+                  const related = await processOutputBBReferences(
```

Comment:
One thing to consider here is performance if we have a lot of rows/links, this is inside a double loop, this await could massively delay things - I wonder if there is a better way we can manage this (like bulking the requests in someway?)

Reply:
As discussed, simplified to use a formula if required